### ISSUE TITLE
False positive for WPS528

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1042]
# Bug report

## What's wrong

False positive for `WPS528`. Because we allow to use mutation inside assignment.
And do not require to use `.items()` here. 

Code:

```python
class MyClass(DjangoForm):
    def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         for field_name in self.fields:
            if field_name not in self._not_required_fields:
                self.fields[field_name].required = True
```

Output:

```
./server/web_app/logics/forms/base.py

  22:9     WPS528 Found implicit `.items()` usage
  for field_name in self.fields:
  ^
```

**Comment by sobolevn** [comment]
Sorry, this one is an invalid issue. This code should be refactored like so:

```python
       for field_name, field in self.fields.item():
            if field_name not in self._not_required_fields:
                field.required = True
```

**Comment by sobolevn** [comment]
But, this code should not raise a violation:

```python
for field_name in self.fields:
    if field_name not in self._not_required_fields:
        self.fields[field_name] += 1
```

All other self-operations should also be fine: `-=`, `*=`, etc
